File: src/entities/ActivityLog.ts
import { Entity, PrimaryGeneratedColumn, Column, ManyToOne, CreateDateColumn } from 'typeorm';
import { User } from './User';
import { Business } from './Business';

@Entity()
export class ActivityLog {
  @PrimaryGeneratedColumn('uuid')
  id: string;

  @Column()
  action: string;

  @Column({ type: 'json', nullable: true })
  metadata: any;

  @ManyToOne(() => User, (user) => user.activityLogs)
  user: User;

  @ManyToOne(() => Business, (business) => business.activityLogs)
  business: Business;

  @CreateDateColumn()
  createdAt: Date;
} 

File: Entities/ActivityLog.cs
using System;

namespace ModularSaaS.Entities
{
    public class ActivityLog
    {
        public Guid Id { get; set; }
        public Guid UserId { get; set; }
        public Guid BusinessId { get; set; }
        public string Action { get; set; }
        public string Metadata { get; set; } // JSON string
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        public User User { get; set; }
        public Business Business { get; set; }
    }
} 